girder/models/model_base.py

```diff
@@ -798,11 +798,26 @@ def load(self, id, level=AccessType.ADMIN, user=None, objectId=True,
         :raises ValidationException: If an invalid ObjectId is passed.
         :returns: The matching document, or None if no match exists.
         """
-        doc = Model.load(self, id=id, objectId=objectId, fields=fields, exc=exc)
+
+        # Ensure we load access and public, these are needed by requireAccess
+        loadFields = None
+        if not force and fields:
+            loadFields = fields[:]
+            loadFields += ['access', 'public']
```

Comment:
It may not matter much, but to avoid duplication we could change to

```
if not force and fields:
    loadFields = list(set(fields) | {'access', 'public'})
```